lib/openapi/api/api_v1.go

```diff
@@ -481,6 +482,39 @@ func (e *Processor) NodeThisMaintenanceGet(c echo.Context, params types.NodeThis
 	return c.JSON(http.StatusOK, params)
 }
 
+func (e *Processor) NodeThisProfilingIndexGet(c echo.Context) error {
+	return e.NodeThisProfilingGet(c, "")
+}
+
+func (e *Processor) NodeThisProfilingGet(c echo.Context, handler string) error {
+	user := c.Get("user")
+	if user.(*types.User).Name != "admin" {
+		c.JSON(http.StatusBadRequest, H{"message": fmt.Sprintf("Only 'admin' can see profiling info")})
+		return fmt.Errorf("Only 'admin' user can see profiling info")
+	}
+
+	switch handler {
+	case "":
+		// Show index if no handler name provided
+		pprof.Index(c.Response().Writer, c.Request())
+	case "allocs", "block", "goroutine", "heap", "mutex", "threadcreate":
+		// PProf usual handlers
+		pprof.Handler(handler).ServeHTTP(c.Response(), c.Request())
+	case "cmdline":
+		pprof.Cmdline(c.Response(), c.Request())
+	case "profile":
+		pprof.Profile(c.Response(), c.Request())
+	case "symbol":
+		pprof.Symbol(c.Response(), c.Request())
+	case "trace":
+		pprof.Trace(c.Response(), c.Request())
+	default:
+		return fmt.Errorf("Unable to find requested profiling handler")
```

Comment:
Currently we'll get a 500 internal server error reported back if you query `/profiling/handler_that_does_not_exist`.  To fix:

```diff
--- a/lib/openapi/api/api_v1.go
+++ b/lib/openapi/api/api_v1.go
@@ -509,6 +509,7 @@ func (e *Processor) NodeThisProfilingGet(c echo.Context, handler string) error {
        case "trace":
                pprof.Trace(c.Response(), c.Request())
        default:
+               c.JSON(http.StatusNotFound, H{"message": fmt.Sprintf("Unable to find requested profiling handler.")})
                return fmt.Errorf("Unable to find requested profiling handler")
        }


```

Reply:
Whoops... slight improvement to the above (no need for `Sprintf` or redundant string text, just put it in a variable):

```diff
diff --git a/lib/openapi/api/api_v1.go b/lib/openapi/api/api_v1.go
index 7e99197..b78473e 100644
--- a/lib/openapi/api/api_v1.go
+++ b/lib/openapi/api/api_v1.go
@@ -509,7 +509,9 @@ func (e *Processor) NodeThisProfilingGet(c echo.Context, handler string) error {
 	case "trace":
 		pprof.Trace(c.Response(), c.Request())
 	default:
-		return fmt.Errorf("Unable to find requested profiling handler")
+		message := "Unable to find requested profiling handler"
+		c.JSON(http.StatusNotFound, H{"message": message})
+		return fmt.Errorf(message)
 	}
 
 	return nil

```

Reply:
Fixed, did the same with error about the admin user in: 07c1304b801d570b1c9de6e010d27576abe9dbf3

Reply:
Awesome, looks great!